### ISSUE TITLE
`array-callback-return` should not trigger for async functions

### ISSUE THREAD

**Issue by gabmontes** [eslint/eslint#7413]
The rule `array-callback-return` checks several array methods are passed a function that return a value. But if you pass an async function, even if the function does not explicitly `return` a value, it will return a promise. I.e. when you want to remove items from a complex schema in a database.

* **ESLint Version:** 2.13.1
* **Node Version:** 6.9.0
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```json
{
  "parser": "babel-eslint",
  "extends": "airbnb"
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
await Promise.all(items.map(async item => {
  await item.parent.removeReference(item)
  await item.remove()
})
```

**What did you expect to happen?**
The rule triggers warning the function does not return a value, which is incorrect.

**What actually happened? Please include the actual, raw output from ESLint.**
`error  Expected to return a value in this function  array-callback-return`

**Comment by gyandeeps** [comment]
Can you try it with latest eslint version and using the default parser?
Since we support `async/await` natively starting with 3.6 release.

**Comment by not-an-aardvark** [comment]
I think this was fixed in https://github.com/eslint/eslint/pull/7172.

**Comment by gabmontes** [comment]
Confirmed. Thanks!!!

**Comment by not-an-aardvark** [comment]
No problem! I'm going to close this issue since it seems to be resolved, but feel free to continue the conversation.
